alibi/models/pytorch/cfrl_models.py

```diff
@@ -0,0 +1,157 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from typing import List
+from alibi.models.pytorch.model import Model
+
+
+class MNISTClassifier(Model):
+    """ Mnist classifier. """
+
+    def __init__(self, output_dim: int) -> None:
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        output_dim
+            Output dimension.
+        """
+        super().__init__()
+
+        self.conv1 = nn.Conv2d(1, 64, kernel_size=(2, 2), padding=1)
+        self.maxpool1 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.dropout1 = nn.Dropout(0.3)
+        self.conv2 = nn.Conv2d(64, 32, kernel_size=(2, 2), padding=1)
+        self.maxpool2 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.dropout2 = nn.Dropout(0.3)
+        self.flatten = nn.Flatten()
+        self.fc1 = nn.Linear(1568, 256)
+        self.fc2 = nn.Linear(256, output_dim)
+
+        # send to device
+        self.to(self.device)
```

Comment:
The MNIST classifier inherits from the `Model` class which sets the device. It just tries to avoid setting the device manually by the user and use the GPU automatically if available. There are not any drawbacks. Probably I was trying to have something more similar with tf.